python/kserve/kserve/protocol/dataplane.py

```diff
@@ -238,18 +238,25 @@ def decode_cloudevent(self, body) -> Tuple[Union[Dict, InferRequest], Dict]:
         decoded_body = body
         attributes = {}
         if isinstance(body, CloudEvent):
-            # Try to decode and parse JSON UTF-8 if possible, otherwise
-            # just pass the CloudEvent data on to the predict function.
-            # This is for the cases that CloudEvent encoding is protobuf, avro etc.
-            try:
-                decoded_body = orjson.loads(body.data.decode('UTF-8'))
-                attributes = body._get_attributes()
-            except (orjson.JSONDecodeError, UnicodeDecodeError) as e:
-                # If decoding or parsing failed, check if it was supposed to be JSON UTF-8
-                if "content-type" in body._attributes and \
-                        (body._attributes["content-type"] == "application/cloudevents+json" or
-                         body._attributes["content-type"] == "application/json"):
-                    raise InvalidInput(f"Failed to decode or parse binary json cloudevent: {e}")
+            attributes = body._get_attributes()
+            avro_content_types = ("application/avro", "application/cloudevents+avro", "application/x-avro-binary",
```

Comment:
I think the content type is not limited to avro, it is for other binary encoded types as well such as protobuf. see more details on this issue https://github.com/cloudevents/spec/issues/520